internal/collector/config_test.go

```diff
@@ -56,12 +62,43 @@ processors:
   batch/200ms:
     timeout: 200ms
   groupbyattrs/compact: {}
+  resourcedetection:
+    detectors: []
+    override: false
+    timeout: 30s
 receivers: {}
 service:
   extensions: []
   pipelines: {}
 `)
 	})
+
+	t.Run("Detectors", func(t *testing.T) {
+		var spec *v1beta1.InstrumentationSpec
+		require.UnmarshalInto(t, &spec, `{
+			config: {
+				detectors: [
+					{ name: gcp },
+					{ name: aks, attributes: { k8s.cluster.name: true } },
+				],
+			},
+		}`)
+
+		result, err := NewConfig(spec).ToYAML()
+		assert.NilError(t, err)
+		assert.Assert(t, cmp.Contains(result, `
+  resourcedetection:
+    aks:
+      resource_attributes:
+        k8s.cluster.name:
+          enabled: true
+    detectors:
+    - gcp
+    - aks
+    override: false
+    timeout: 30s
```

Comment:
Huh, so this form lets us avoid specifying the resourcedetection sources like `resourcedetection/ec2`.

(hmmm, for a moment I had `/aws` there before double-checking -- not here, but maybe in the docs we want to say what the allowable options are or say that there are only a set list and send people to the otel docs.

Reply:
Yeah, this component handles lots of ... sources. 🤔 Now that I think about it, that seems unusual because it means you must include/compile support for multiple clouds, even if you don't use them.